### PR TITLE
Adding Rewind Functionality

### PR DESCRIPTION
**Pull request by rbingen3**
Rewind Functionality based on information on Piazza. UI can call rewindSimulation() for each click of the rewind button.

### CODE REVIEW

File: src/Simulation.java [hunk 1]
```diff
@@ -21,12 +22,28 @@ public Simulation (String filename, int iterations)
...
+	{
+		if (snapshots.size() > 0)
+		{
+			buses = snapshots.get(0).getBuses();
+			stops = snapshots.get(0).getStops();
+			events.add(snapshots.get(0).getEvent());
+			currentTime = snapshots.get(0).getEvent().time;
+			sortEvents();
+			snapshots.remove(0);
+		}
```
Comment (bwang-ith) [inline src/Simulation.java]: else?

File: src/Snapshot.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import java.util.List;
+
+public class Snapshot 
+{
+	private List<Bus> buses;
```
Comment (bwang-ith) [inline src/Snapshot.java]: Don't we need to capture more state e.g. routes, depots, passengers? What if a route has changed within the last three events? Also, do you need this class? Sorry if I haven't this totally through, but could you just instantiate the Simulation class to use as snapshots?

File: src/Simulation.java (hunk 1 above)
Comment (Neoender) [reply src/Simulation.java]: The else is it wouldn't do anything, but the else can be updated when the UI is done to add whatever response we want.

File: src/Snapshot.java (hunk 2 above)
Comment (Neoender) [reply src/Snapshot.java]: Based on the office hours and Piazza posts we only need to rewind the events, location of the bus, and the passengers. This handles all three as this implementation does not have a passenger object, the buses and stops know how many people they have. However I will add the routes to the Snapshot to make sure they are in sync even though they are not required to be rewound.

As for if Snapshot should be a new class, I'll say yes it should. In my mind the Simulation.java manages the whole application and therefore new UIs can just be thrown over it without any non-ui logic needing to be added, including keeping track of which simulation instance is being used. I think either will work, but this fits better with the current architecture.

**Pull request closed without merging**